langchain4j-oracle/src/main/java/dev/langchain4j/data/document/splitter/oracle/OracleDocumentSplitter.java

```
@@ -0,0 +1,98 @@
+package dev.langchain4j.data.document.splitter.oracle;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.DocumentSplitter;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.segment.TextSegment;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OracleDocumentSplitter implements DocumentSplitter {
+
+    private static final Logger log = LoggerFactory.getLogger(OracleDocumentSplitter.class);
+
+    private static final String INDEX = "index";
+
+    private final Connection conn;
+    private final String pref;
+
+    public OracleDocumentSplitter(Connection conn, String pref) {
+        this.conn = conn;
+        this.pref = pref;
+    }
+
+    @Override
+    public List<TextSegment> split(Document document) {
+        List<TextSegment> segments = new ArrayList<>();
+        try {
+            String[] parts = split(document.text());
+            int index = 0;
+            for (String part : parts) {
+                segments.add(createSegment(part, document, index));
+                index++;
+            }
+        } catch (SQLException | JsonProcessingException e) {
+            String message = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
+            log.warn("Failed to summarize '{}': {}", pref, message);
```

Is it ok to just log a warning and return an empty List?

---

The interface doesn't provide for an exception

    public List<TextSegment> split(Document document) {

Perhaps we should throw a runtime exception?

        try {
          ...
        } catch (SQLException | JsonProcessingException e) {
          throw new RuntimeException(e);
        }

---

Yes, particularly if there's no practical reason on trying to recover from this condition (not sure if it does, just stating the reasoning). And yes for a RuntimeException or subclass.. You could also add a message with the original exception. Another idea would be to check other implementations in case the library already has a RuntimeException subclass we can use.

---

Changed to throw a RuntimeException